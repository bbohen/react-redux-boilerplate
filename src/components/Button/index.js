import styled from 'styled-components';

const Button = styled.button`
  border: none;
  background: ${props => props.theme.colors.primary};
  color: ${props => props.theme.colors.light};
  padding: 1em;
`;

export default Button;
