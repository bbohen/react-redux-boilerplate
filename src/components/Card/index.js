import PropTypes from 'prop-types';
import React from 'react';

import Img from './Img';
import Wrapper from './Wrapper';

const Card = ({ onClick, imageUrl, index }) => {
  const handleClick = onClick.bind(null, index);
  return (
    <Wrapper
      onClick={handleClick}
    >
      <Img
        alt=""
        src={imageUrl}
      />
    </Wrapper>
  );
};

Card.propTypes = {
  index: PropTypes.number.isRequired,
  imageUrl: PropTypes.string,
  onClick: PropTypes.func,
};

Card.defaultProps = {
  onClick: () => false,
  imageUrl: '',
};

export default Card;
