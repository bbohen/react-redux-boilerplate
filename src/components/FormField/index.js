import PropTypes from 'prop-types';
import React from 'react';
import Feedback from './Feedback';
import Input from './Input';
import Label from './Label';
import Wrapper from './Wrapper';

const FormField = ({
  input,
  label,
  placeholder,
  type,
  meta: { touched, error, warning },
}) => {
  const showError = touched && error;
  const showWarning = touched && warning;

  return (
    <Wrapper>
      {label &&
      <Label
        error={showError}
        warning={showWarning}
      >
        {label}
      </Label> }
      <Input
        error={showError}
        warning={showWarning}
        placeholder={placeholder}
        type={type}
        {...input}
      />
      {((showError &&
        <Feedback
          error={showError}
          warning={showWarning}
        >
          {error}
        </Feedback>) ||
      (showWarning &&
        <Feedback
          error={showError}
          warning={showWarning}
        >
          {warning}
        </Feedback>))}
    </Wrapper>
  );
};

FormField.propTypes = {
  input: PropTypes.shape().isRequired,
  label: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.boolean,
    error: PropTypes.string,
    warning: PropTypes.string,
  }),
  placeholder: PropTypes.string,
  type: PropTypes.string,
};

FormField.defaultProps = {
  label: undefined,
  placeholder: '',
  type: 'text',
  meta: {
    touched: false,
  },
};

export default FormField;
