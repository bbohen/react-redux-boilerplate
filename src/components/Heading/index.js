import styled from 'styled-components';

const Heading = styled.h2`
  color: ${props => props.theme.colors.secondary};
  margin: 0;
  font-family: 'Quicksand';
`;

export default Heading;
