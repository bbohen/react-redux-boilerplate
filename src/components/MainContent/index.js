import PropTypes from 'prop-types';
import React from 'react';

import Wrapper from './Wrapper';
import Content from './Content';

const MainContent = ({ children }) => (
  <Wrapper>
    <Content>
      {children}
    </Content>
  </Wrapper>
);

MainContent.propTypes = {
  children: PropTypes.element,
};

MainContent.defaultProps = {
  children: [],
};

export default MainContent;
