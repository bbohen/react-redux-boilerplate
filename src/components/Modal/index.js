import React, { PropTypes } from 'react';

import Wrapper from './Wrapper';
import Content from './Content';

const Modal = ({ children, onClick }) =>
  (<Wrapper onClick={onClick}>
    <Content>
      {children}
    </Content>
  </Wrapper>);

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  onClick: PropTypes.func,
};

Modal.defaultProps = {
  onClick: () => false,
};

export default Modal;
