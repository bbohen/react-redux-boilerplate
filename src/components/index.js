export Button from './Button';
export Card from './Card';
export FormField from './FormField';
export Heading from './Heading';
export Modal from './Modal';
export NotFound from './NotFound';
