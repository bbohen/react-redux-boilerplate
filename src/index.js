import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import { MainContent, NotFound } from 'components';
import { App, Cats, Humans, FormWithCats, Home, Navigation } from 'containers';
import initializeStore from './redux/init';

import './globalStyles';

require('offline-plugin/runtime').install();

const store = initializeStore();

const Boilerplate = () => (
  <Provider store={store}>
    <BrowserRouter>
      <App>
        <Navigation />
        <MainContent>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/cats" component={Cats} />
            <Route path="/form" component={FormWithCats} />
            <Route path="/humans" component={Humans} />
            <Route component={NotFound} />
          </Switch>
        </MainContent>
      </App>
    </BrowserRouter>
  </Provider>
);

ReactDOM.render(
  <Boilerplate />,
  document.getElementById('mount'),
);

export default Boilerplate;
