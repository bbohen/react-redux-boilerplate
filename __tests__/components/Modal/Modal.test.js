import React from 'react';
import renderer from 'react-test-renderer';

import { Modal } from '../../../src/components';

const mockFunction = jest.fn();

describe('<Modal />', () => {
  it('renders a child component', () => {
    const component = renderer.create(<Modal>
      <div>A child component</div>
    </Modal>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('defaults when given no click event', () => {
    const component = renderer.create(<Modal>
      <div>A child component</div>
                                      </Modal>);
    let tree = component.toJSON();

    tree.props.onClick();
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('executes a passed click event', () => {
    const component = renderer.create(<Modal onClick={mockFunction}>
      <div>A child component</div>
                                      </Modal>);
    let tree = component.toJSON();

    tree.props.onClick();
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    expect(mockFunction.mock.calls.length).toBe(1);
  });
});
