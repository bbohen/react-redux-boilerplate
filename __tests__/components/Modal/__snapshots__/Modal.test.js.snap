// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> defaults when given no click event 1`] = `
<div
  className="sc-htoDjs kpoONz"
  onClick={[Function]}
>
  <div
    className="sc-dnqmqq CmCVf"
  >
    <div>
      A child component
    </div>
  </div>
</div>
`;

exports[`<Modal /> executes a passed click event 1`] = `
<div
  className="sc-htoDjs kpoONz"
  onClick={[Function]}
>
  <div
    className="sc-dnqmqq CmCVf"
  >
    <div>
      A child component
    </div>
  </div>
</div>
`;

exports[`<Modal /> renders a child component 1`] = `
<div
  className="sc-htoDjs kpoONz"
  onClick={[Function]}
>
  <div
    className="sc-dnqmqq CmCVf"
  >
    <div>
      A child component
    </div>
  </div>
</div>
`;
